python/GafferUI/SpreadsheetUI/_PlugTableView.py

```diff
@@ -432,65 +432,64 @@ def __plugMetadataChanged( self, plug, key, reason ) :
+	# specified event, or `None` if dropping is not possible.
+	def __dropFunction( self, event ) :
 
 		if Gaffer.MetadataAlgo.readOnly( self._qtWidget().model().rowsPlug() ) :
-			return False
-
-		if not isinstance( event.data, ( Gaffer.Plug, IECore.Data ) ) :
-			return False
-
-		return True
-
-	def __dragMove( self, widget, event ) :
+			return None
 
 		destinationPlug = self.plugAt( event.line.p0 )
-		selectionModel = self._qtWidget().selectionModel()
-
 		if destinationPlug is None :
-			selectionModel.clear()
-			return
+			return None
 
-		select = False
 		if isinstance( event.data, IECore.Data ) :
-			select = _ClipboardAlgo.canPasteCells( event.data, [ [ destinationPlug ] ] )
-		else :
+			if _ClipboardAlgo.canPasteCells( event.data, [ [ destinationPlug ] ] ) :
+				return functools.partial(
+					_ClipboardAlgo.pasteCells,
+					event.data, [ [ destinationPlug ] ],
+					self.ancestor( GafferUI.PlugValueWidget ).getContext().getTime()
+				)
+		elif isinstance( event.data, Gaffer.Plug ) :
 			sourcePlug, targetPlug = self.__connectionPlugs( event.data, destinationPlug )
-			select = self.__canConnect( sourcePlug, targetPlug )
+			if self.__canConnect( sourcePlug, targetPlug ) :
+				return functools.partial( targetPlug.setInput, sourcePlug )
+
+		return None
+
+	def __dragEnter( self, widget, event ) :
+
+		return self.__dropFunction( event ) is not None
 
-		if select :
-			selectionModel.select(
-				self._qtWidget().model().indexForPlug( destinationPlug ),
+	def __dragMove( self, widget, event ) :
+
+		if self.__dropFunction( event ) is None :
+			self._qtWidget().selectionModel().clear()
+		else :
+			self._qtWidget().selectionModel().setCurrentIndex(
+				self._qtWidget().model().indexForPlug( self.plugAt( event.line.p0 ) ),
 				QtCore.QItemSelectionModel.SelectCurrent
 			)
-		else :
-			selectionModel.clear()
+
+		return True
 
 	def __dragLeave( self, widget, event ) :
 
 		self._qtWidget().selectionModel().clear()
+		return True
 
 	def __drop( self, widget, event ) :
 
-		destinationPlug = self.plugAt( event.line.p0 )
+		# Update the selection in case the drag position changed
+		# since last `__dragMove` (not sure this is necessary).
+		self.__dragMove( widget, event )
 
-		if isinstance( event.data, IECore.Data ) :
-			if not _ClipboardAlgo.canPasteCells( event.data, [ [ destinationPlug ] ] ) :
-				return False
-			with Gaffer.UndoScope( destinationPlug.ancestor( Gaffer.ScriptNode ) ) :
-				context = self.ancestor( GafferUI.PlugValueWidget ).getContext()
-				_ClipboardAlgo.pasteCells( event.data, [ [ destinationPlug ] ], context.getTime() )
-		else :
-			sourcePlug, targetPlug = self.__connectionPlugs( event.data, destinationPlug )
-			if not self.__canConnect( sourcePlug, targetPlug ) :
-				return False
-			with Gaffer.UndoScope( targetPlug.ancestor( Gaffer.ScriptNode ) ) :
-				targetPlug.setInput( sourcePlug )
+		dropFunction = self.__dropFunction( event )
+		if dropFunction is None :
+			return True
```

Comment:
What does the return value of __drop do?  It looks like in the case of `not canPasteCells`, we previously returned False, and now return True?  And we don't set the return value at all if there actually is a dropFunction?   Does this return value even do anything?

Reply:
The return value is used to shortcut the evaluation of any other slots connected to `dropSignal()`, matching the behaviour of pretty much all the other UI signals. Our expectation here is that we are the sole handler, so I'm returning True so that there's no loophole whereby another handler can sneak in in certain circumstances (having multiple drag handlers isn't  recommended due to the difficulty of getting all the interactions between dragEnter/DragMove/drop working correctly).

Reply:
If there is a dropFunction, we also return True, at line 552.

Reply:
Ah, makes sense.  Sorry I missed that `return True`, github formatting tripped me up I think.